remix-analyzer/src/solidity-analyzer/modules/gasCosts.ts

```diff
@@ -1,72 +1,96 @@
 import { default as category } from './categories'
 import { default as algorithm } from './algorithmCategories'
-import  AbstractAst from './abstractAstView'
-import { ModuleAlgorithm, ModuleCategory, ReportObj, CompilationResult, CompiledContractObj, CompiledContract, VisitFunction, AnalyzerModule} from './../../types'
-
-interface VisitedContract {
-  name: string
-  object: CompiledContract
-  file: string
-}
+import { getFunctionDefinitionName, helpers } from './staticAnalysisCommon'
+import { ModuleAlgorithm, ModuleCategory, ReportObj, CompilationResult, CompiledContract, AnalyzerModule, 
+  FunctionDefinitionAstNode, VariableDeclarationAstNode } from './../../types'
 
 export default class gasCosts implements AnalyzerModule {
   name: string = `Gas costs: `
   description: string = `Too high gas requirement of functions`
   category: ModuleCategory = category.GAS
   algorithm: ModuleAlgorithm = algorithm.EXACT
 
-  abstractAst: AbstractAst = new AbstractAst()
-  visit: VisitFunction = this.abstractAst.build_visit((node: any) => false)
+  warningNodes: any[] = []
+  visit (node: FunctionDefinitionAstNode | VariableDeclarationAstNode): void {
+    if ((node.nodeType === 'FunctionDefinition' && node.kind !== 'constructor' && node.implemented) || 
+    (node.nodeType === 'VariableDeclaration' && node.stateVariable && node.visibility === 'public')) 
```

Comment:
Maybe there could be an helper (or that already exist?) for checking if a state variable will be turned into a getter.

Reply:
This has been added